register-form/src/components/RegForm.vue

```diff
@@ -0,0 +1,186 @@
+
+<template>
+  <div>
+    <label>First name:</label><br>
+    <input :class="{invalid: !valids.firstName & !firstTime}" type="text" v-model="inputs.firstName" >
+    <br>
+    <span class="error" v-show="!valids.firstName & !firstTime">
+          *First name must have atleast one letter.</span>
+    <br>
+
+    <label>Last name:</label><br>
+    <input :class="{invalid: !valids.lastName & !firstTime}" type="text" v-model="inputs.lastName">
+        <br>
+    <span class="error" v-show="!valids.lastName & !firstTime">
+          *Last name must have atleast one letter.</span>
+    <br>
+
+    <label >Gender:</label> <br>
+    <select :class="{invalid: !valids.gender & !firstTime}" v-model="inputs.gender">
+      <option value="" selected>Gender</option>
+      <option value="male">Male</option>
+      <option value="female">Female</option>
+      <option value="other">other</option>
+    </select> 
+    <br>
+    <span class="error" v-show="!valids.gender & !firstTime">*Must chose a gender.</span>
+    <br>  
+
+
+    <label>Email:</label><br>
+    <input :class="{invalid: !valids.email & !firstTime}" type="text" v-model="inputs.email">
+    <br>
+    <span class="error" v-show="!valids.email & !firstTime">*Email must have the form: 'X@X.X'.</span>
+    <br>
+
+    <label>Phone number:</label><br>
+    <input :class="{invalid: !valids.phoneNum & !firstTime}" type="text" v-model="inputs.phoneNum">
+    <br>
+    <span class="error" v-show="!valids.phoneNum & !firstTime">*Phone number must have 10 digits.</span>
+    <br>
+    
+    <p>Chose your power animal:</p>
+    <label v-for="img in images" :key="img.val" :class="{invalidAvatar: !valids.avatar & !firstTime}">
+      <input type="radio" :value="img.val" v-model="inputs.avatar">
+      <img class="radio-img" :src="img.src" :alt="img.val"> 
+    </label>
+    <br>
+    <span class="errorAvatar" v-show="!valids.avatar & !firstTime">*Must chose an avatar.</span>
+    <br>
+
+    <input type="button" class="submit" value="Submit" @click="submitForm()">
```

Comment:
the syntax `@click="submitForm"` (without the parenthesis) is better here and in general - it is like the difference between `addEventListener('click', submitForm)` and `addEventListener('click', () => {submitForm();})`  - the latter don;t allow us to pass the event to the function and instead of binding the actual function to the event it binds a wrapper of the function which can interfere with reactivity sometimes